python/beeai_framework/cancellation.py

```diff
@@ -66,7 +66,7 @@ def timeout(cls, duration: float) -> "AbortSignal":
 
     def throw_if_aborted(self) -> None:
         if self._aborted:
-            raise AbortError(self._reason)
+            raise AbortError(self.reason)
```

Comment:
why?

Reply:
`AbortError` expects a str but `self._reason` can be None whereas `self.reason` handles the None case